nav2_smac_planner/src/node_lattice.cpp

```diff
@@ -81,9 +145,47 @@ LatticeMetadata LatticeMotionTable::getLatticeMetadata(const std::string & latti
   // TODO(Matt) from this file extract and return the number of angle bins and
   // turning radius in global coordinates, respectively.
   // world coordinates meaning meters, not cells
-  return {0 /*num bins*/, 0 /*turning rad*/};
+  return LatticeMetadata();
+}
+
+void fromJsonToMetaData(const nlohmann::json &j, LatticeMetadata &latticeMetadata)
+{
+  j.at("turningRadius").get_to(latticeMetadata.min_turning_radius);
+  j.at("stepDistance").get_to(latticeMetadata.step_distance);
+  j.at("gridSeparation").get_to(latticeMetadata.grid_separation);
+  j.at("maxLength").get_to(latticeMetadata.max_length);
+  j.at("numberOfHeadings").get_to(latticeMetadata.number_of_headings);
+  j.at("outputFile").get_to(latticeMetadata.output_file);
+  j.at("headingAngles").get_to(latticeMetadata.heading_angles);
+  j.at("numberOfTrajectories").get_to(latticeMetadata.number_of_trajectories);
 }
 
+void fromJsonToPose(const nlohmann::json &j, MotionPose &pose)
+{
+  pose._x = j[0]; 
+  pose._y = j[1]; 
+  pose._theta = j[2];
+}
+
+void fromJsonToMotionPrimitive(const nlohmann::json &j, MotionPrimitive &motionPrimitive)
+{
+  j.at("trajectoryId").get_to(motionPrimitive.trajectory_id);
+  j.at("startAngle").get_to(motionPrimitive.start_angle);
+  j.at("endAngle").get_to(motionPrimitive.end_angle);   
+  j.at("radius").get_to(motionPrimitive.radius);
+  j.at("trajectoryLength").get_to(motionPrimitive.trajectory_length);
+  j.at("arcLength").get_to(motionPrimitive.arc_length);
+  j.at("straightLength").get_to(motionPrimitive.straight_length);
+
+  for(auto& jsonPose : j["poses"])
```

Comment:
just in styling for the rest of the package, can you do a C-style for loop instead?

Reply:
Done